src/Kaleidoscope.h

```diff
@@ -147,21 +133,39 @@ class Kaleidoscope_ {
   typedef Key(*eventHandlerHook)(Key mappedKey, byte row, byte col, uint8_t keyState);
   static eventHandlerHook eventHandlers[HOOK_MAX];
 
-  static void replaceEventHandlerHook(eventHandlerHook oldHook, eventHandlerHook newHook);
-  static void appendEventHandlerHook(eventHandlerHook hook);
-  static void useEventHandlerHook(eventHandlerHook hook);
+  static void replaceEventHandlerHook(eventHandlerHook oldHook, eventHandlerHook newHook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.onKeyswitchEvent(...) instead.")));
+  static void appendEventHandlerHook(eventHandlerHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.onKeyswitchEvent(...) instead.")));
+  static void useEventHandlerHook(eventHandlerHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.onKeyswitchEvent(...) instead.")));
 
   typedef void (*loopHook)(bool postClear);
   static loopHook loopHooks[HOOK_MAX];
 
-  static void replaceLoopHook(loopHook oldHook, loopHook newHook);
-  static void appendLoopHook(loopHook hook);
-  static void useLoopHook(loopHook hook);
+  static void replaceLoopHook(loopHook oldHook, loopHook newHook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.beforeEachCycle(), .beforeReportingState(...) or .afterEachCycle(...) instead.")));
+  static void appendLoopHook(loopHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.beforeEachCycle(), .beforeReportingState(...) or .afterEachCycle(...) instead.")));
+  static void useLoopHook(loopHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.beforeEachCycle(), .beforeReportingState(...) or .afterEachCycle(...) instead.")));
 
   static bool focusHook(const char *command);
 };
 
-extern Kaleidoscope_ Kaleidoscope;
+extern kaleidoscope::Kaleidoscope_ Kaleidoscope;
+
+} // namespace kaleidoscope
+
+// For compatibility reasons we enable class Kaleidoscope_ also to be available
+// in global namespace.
+//
+typedef kaleidoscope::Kaleidoscope_  Kaleidoscope_;
+
+// For compatibility reasons we enable the global variable Kaleidoscope
+// in global namespace.
+//
+extern kaleidoscope::Kaleidoscope_ &Kaleidoscope;
```

Comment:
Not urgent:

Damn! It just occured to me that we could have fixed this with a [using declaration](https://godbolt.org/g/LKKnMZ) to generate a perfect alias and thereby saving 2 bytes of RAM. 
```
using kaleidoscope::Kaleidoscope;
```
If someone changes this, don't forget to remove the instanciation of `&Kaleidoscope` in `kaleidoscope.cpp`.

Reply:
Hah! Nice. It also appears to have saved us a few (20) bytes of PROGMEM too.

Reply:
20 bytes! For a simple reference. I didn't expect that much.